src/main/java/com/sucy/skill/api/player/PlayerData.java

```diff
@@ -314,9 +315,21 @@ public HashMap<String, Integer> getInvestedAttributes() {
         return new HashMap<>(attributes);
     }
 
+    /**
+     * Retrieves a map of all attributes upgrade stages.
+     * This doesn't count base attributes from classes or
+     * bonus attributes from effects. Modifying the map will
+     * not change actual player attributes.
+     *
+     * @return attribute upgrade stages
+     */
+    public HashMap<String, Integer> getInvestedAttributesStages() {
+        return new HashMap<>(attrUpStages);
+    }
+
     /**
      * Gets the number of attribute points the player has
-     * between invested and bonus sources.
+     * from invested and bonus sources.
```

Comment:
Yeah, I guess there was a typo cuz there is a sum so => 
it's (_from_) total = invested + bonus, not the substraction (_between_) invested - bonus

Reply:
Shall we keep from for clarity maybe?

Reply:
Yeah, no reason why not.